apps/sales/views.py

```diff
@@ -22,7 +23,15 @@ class SaleOrderView(viewsets.ModelViewSet):
     serializer_class = SaleOrderSerializer
     
     def list(self, request, *args, **kwargs):
-        return list_all_objects(self, request, *args, **kwargs)
+        summary = request.query_params.get('summary', 'false').lower() == 'true'
+        if summary:
+            sale_order = self.filter_queryset(self.get_queryset())
+            data = SaleOrderOptionsSerializer.get_sale_order_summary(sale_order)
+            result = Response(data, status=status.HTTP_200_OK)
```

Comment:
you should be using build_response util method here.

Reply:
Now using build_response